chains/ethereum/contract/LockRedeemValidatorFees.sol

```diff
@@ -0,0 +1,222 @@
+        require(amount_ > 0, "amount should be bigger than 0");
+        require(redeemRequests[msg.sender].until < block.number, "request is locked, not available");
+
+        redeemRequests[msg.sender].isCompleted = false;
+        redeemRequests[msg.sender].signature_count = uint256(0);
+        redeemRequests[msg.sender].recipient = msg.sender;
+        redeemRequests[msg.sender].amount = amount_ ;
+        redeemRequests[msg.sender].redeemFee = msg.value;
+        redeemRequests[msg.sender].until = block.number + LOCK_PERIOD;
+
+        emit RedeemRequest(redeemRequests[msg.sender].recipient,redeemRequests[msg.sender].amount);
+    }
+
+    //function called by Validators using protocol to sign on RedeemRequests
+    function sign(uint amount_, address payable recipient_) public isActive onlyValidator {
+        uint startGas = gasleft();
+        require(!redeemRequests[recipient_].isCompleted, "redeem request is completed");
+        require(redeemRequests[recipient_].amount == amount_,"redeem amount is different" );
+        require(!redeemRequests[recipient_].votes[msg.sender]);
+        require(redeemRequests[recipient_].until > block.number, "request has expired");
+        // update votes
+        redeemRequests[recipient_].votes[msg.sender] = true;
+        redeemRequests[recipient_].signature_count += 1;
+        //validators[msg.sender].validatorFee = validators[msg.sender].validatorFee + tx.gasprice * avg_gas;
+
+        // if threshold is reached then transfer the amount
+        if (redeemRequests[recipient_].signature_count >= votingThreshold ) {
+            (bool success, ) = redeemRequests[recipient_].recipient.call.value((redeemRequests[recipient_].amount))("");
+            require(success, "Transfer failed.");
+            redeemRequests[recipient_].amount = 0;
+            redeemRequests[recipient_].isCompleted = true;
+        }
+
+        // Trasaction Cost is a an average trasaction cost .
+        // additionalGasCost is the extra gas used by the lines of code after gas calculation is done.
+        // This is an approximate calcualtion and actuall cost might vary sightly .
+        uint gasUsed = startGas - gasleft() + trasactionCost + additionalGasCost ;
+        validators[msg.sender].validatorFee = gasUsed;
+        uint gasFee = gasUsed * tx.gasprice;
+        (bool success, ) = msg.sender.call.value(gasFee)("");
+        require(success, "Transfer back to validator failed");
+        redeemRequests[recipient_].redeemFee -= gasFee;
+        emit ValidatorSignedRedeem(recipient_, msg.sender, amount_,gasUsed);
+        // if ( redeemRequests[recipient_].isCompleted == true ) {
+        //      for (uint i = 0; i < initialValidatorList.length; i++) {
+        //      (bool success, ) = initialValidatorList[i].call.value(validators[initialValidatorList[i]].validatorFee)("");
+        //      require(success, "Transfer failed.");
+        // }
+    }
```

Comment:
the fee is not transferred to validator?

Reply:
the transfer is on line 164 , I am using msg.sender instead of the validator list , because the ether is sent back for every signature